### PR TITLE
[MAINTENANCE] Replace Sphinx with new Typo3 documentation variant

### PR DESCRIPTION
**Pull request by markusweigelt**
> Since the beginning of 2024 you can switch to the new, PHP-based reStructured Text rendering to try it out. The new rendering will become mandatory in August 2024.

https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/Migration/Index.html

What does PR do?
- migrate legacy Sphinx to PHP based rendering
- adjust developer documentation
- remove legacy code and files
- Rename images cause images could not be assigned to new documentation
- GitHub workflow to check documentation files for errors and warning
- Bugfixes e.g. User Manual -> Indexing Documents -> Reindex collections table

More informations:
https://docs.typo3.org/other/t3docs/render-guides/main/en-us/Introduction/Index.html

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: test documentation
```
Comment (sebastian-meyer) [inline .github/workflows/documentation.yml]:
```suggestion
name: Build Documentation
```

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+on: [ push, pull_request ]
+
+jobs:
+    tests:
+        name: documentation
```
Comment (sebastian-meyer) [inline .github/workflows/documentation.yml]:
```suggestion
        name: Documentation
```
The names are shown on GitHub, so we should make them pretty! ;o)

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -124,50 +124,68 @@ docker-compose down
...
+```bash
+composer docs:build
+```
 
-To preview the rendered output and automatically rebuild documentation on changes, you may spawn a local server. This supports auto-refresh and is faster than the official preview build, but omits some features such as syntax highlighting.
+Take a look at the documentation by opening the file `Index.html` in the folder
+`Documentation-GENERATED-temp` in your browser.
 
-This requires Python 2 to be installed.
+### Provide via Http Server (optional)
```
Comment (sebastian-meyer) [inline DEVELOPMENT.md]:
```suggestion
### Provide via HTTP Server (optional)
```

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -124,50 +124,68 @@ docker-compose down
...
 
-```bash
-# First start: Setup Sphinx in a virtualenv
-composer docs:setup
+Starts the http server and mounts the mandatory directory `Documentation-GENERATED-temp`.
```
Comment (sebastian-meyer) [inline DEVELOPMENT.md]:
```suggestion
Starts the HTTP server and mounts the mandatory directory `Documentation-GENERATED-temp`.
```

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -124,50 +124,68 @@ docker-compose down
...
+```bash
+composer docs:start
 ```
 
-By default, the output is served to http://127.0.0.1:8000.
+Take a look at the documentation by opening <http://localhost:8000>
+in your browser.
 
-> The setup and serve commands are defined in [Build/Documentation/sphinx.sh](./Build/Documentation/sphinx.sh).
+The server runs in detached mode, so you will need to stop the http server manually.
```
Comment (sebastian-meyer) [inline DEVELOPMENT.md]:
```suggestion
The server runs in detached mode, so you will need to stop it manually.
```

File: Documentation/Plugins/Index.rst [hunk 6]
```diff
@@ -283,13 +283,14 @@ The collection plugin shows one collection, all collections or selected collecti
  - :Property:
        targetPid
    :Data Type:
-       :ref:`t3tsref:data-type-page-id`
+       `t3tsref:data-type-page-id`
+   :
```
Comment (sebastian-meyer) [inline Documentation/Plugins/Index.rst]: This `:` looks weird. Is this a typo or intended?

File: Documentation/guides.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd" links-are-relative="true">
+  <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension" project-home="https://kitodo.org" project-contact="mailto:contact@kitodo.org" project-repository="https://github.com/kitodo/kitodo-presentation" project-issues="https://github.com/kitodo/kitodo-presentation/issues" edit-on-github-branch="master" edit-on-github="kitodo/kitodo-presentation" typo3-core-preferred="stable"/>
+  <project title="Kitodo.Presentation" release="4.0" copyright="since 2017 by Kitodo Release Management Team"/>
```
Comment (sebastian-meyer) [inline Documentation/guides.xml]:
```suggestion
  <project title="Kitodo.Presentation" release="5.0" copyright="since 2017 by Kitodo Release Management Team"/>
```

File: composer.lock
Comment (sebastian-meyer) [inline composer.lock:1]: Why did you remove `composer.lock`?

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

File: composer.lock
Comment (markusweigelt) [reply composer.lock:1]: I didn't understand why the composer.lock file isn't ignored, so it was just a suggestion. Why is it versioned?

Comment (sebastian-meyer) [reply composer.lock:1]: We provide a `composer.lock` to have a common known working state for all the components (especially for testing). It also doesn't cause any issues under "normal" circumstances, because afaik the lock file is ignored when installing Kitodo.Presentation in an already running TYPO3 environment by `composer require kitodo/presentation`.

File: Documentation/Plugins/Index.rst (hunk 6 above)
Comment (markusweigelt) [reply Documentation/Plugins/Index.rst]: Currently, unresolved references are not displayed with the new rendering method. I have already opened an issue for this at https://github.com/TYPO3-Documentation/render-guides/issues/580. Although we could set a default value, doing so would require changing the test level of the documentation workflow from "Fail on Warning" to "Fail on Error." Therefore, I have converted these references into plain text. These references are also unresolved in the current documentation and rendered as plain text.

Comment (sebastian-meyer) [reply Documentation/Plugins/Index.rst]: My question was about the `:` in line 287. :)

File: composer.lock
Comment (sebastian-meyer) [reply composer.lock:1]: Could you please restore the `composer.lock` then? Thanks!

File: Documentation/Plugins/Index.rst (hunk 6 above)
Comment (markusweigelt) [reply Documentation/Plugins/Index.rst]: Oops, I thought something wasn't resolved correctly. Yes, that's not intended.

**Review by sebastian-meyer — APPROVED** [review]

**Comment by sebastian-meyer** [comment]
Thank you very much! And congratulations on your first contribution to Kitodo.Presentation!
